Makefile

```diff
@@ -0,0 +1,42 @@
+# This Makefile is meant to be used by people that do not usually work
+# with Go source code. If you know what GOPATH is then you probably
+# don't need to bother with make.
+
+GOBIN = $(shell pwd)/build/bin
+GO ?= latest
+
+
+swarm:
+	build/env.sh go run build/ci.go install ./cmd/swarm
+	@echo "Done building."
+	@echo "Run \"$(GOBIN)/swarm\" to launch swarm."
+
+all:
+	build/env.sh go run build/ci.go install
+
+test: all
+	build/env.sh go run build/ci.go test
+
+lint: ## Run linters.
+	build/env.sh go run build/ci.go lint
+
+clean:
+	./build/clean_go_build_cache.sh
+	rm -fr build/_workspace/pkg/ $(GOBIN)/*
+
+# The devtools target installs tools required for 'go generate'.
+# You need to put $GOBIN (or $GOPATH/bin) in your PATH to use 'go generate'.
+
+devtools:
+	env GOBIN= go get -u golang.org/x/tools/cmd/stringer
+	env GOBIN= go get -u github.com/kevinburke/go-bindata/go-bindata
+	env GOBIN= go get -u github.com/fjl/gencodec
+	env GOBIN= go get -u github.com/golang/protobuf/protoc-gen-go
+	env GOBIN= go install ./cmd/abigen
+	@type "npm" 2> /dev/null || echo 'Please install node.js and npm'
+	@type "solc" 2> /dev/null || echo 'Please install solc'
+	@type "protoc" 2> /dev/null || echo 'Please install protoc'
+
+swarm-devtools:
```

Comment:
Why do we need such a target?

Reply:
It was in the previous version over at go-ethereum.
My PR is only to import that functionality into our new repo. Pruning the content seems like another step that I did not feel fully qualified to make.

Reply:
I see that it is just an import, but we literally don't need this whole file. I'd rather if we just don't have a Makefile at all.

All you need to build Swarm is part of the README.md already and it is two commands.

Running the tests is a single command.

Usually you need a Makefile when you add some non-obvious behaviour or you do multiple steps as part of the build and test processes, which we don't currently.

Reply:
:(
I heartily disagree. 
`make swarm` works when `go install` does not.

It works from any directory. It works without GOPATH, it works without GOROOT. It is far easier to use for anyone who is not already a go developer with a fully installed and configured go development environment.

Reply:
Here is what it looks like from my perspective:

sorry we have removed the makefile, you can no longer use `make swarm` to compile swarm.
Instead please create a directory called go, 
in it create a directory called src
in it create a directory called github.com
in it create a directory called ethersphere
move the swarm source into this directory.
Next set up the necessary environment variables for go to work (see golang documentation)
then use `go install` instead of `make`.


and all that why?

Reply:
@homotopycolimit right - it wasn't obvious right from the get-go that the purpose of the Makefile is to build the project for people that don't have a proper Go environment setup - i.e. `GOPATH`, etc.

If you use the go tooling, `go get` builds the proper paths for you, so you don't have to create directories yourself.

In the future we could modify our vendoring strategy and use `go.mod` so that we don't have the requirement for GOPATH.

Until then I agree that having a Makefile is useful, so that it builds `swarm` for users who have just cloned the repo.